hudi-sync/hudi-hive-sync/src/main/java/org/apache/hudi/hive/HiveSyncTool.java

```diff
@@ -103,15 +103,29 @@ public class HiveSyncTool extends HoodieSyncTool implements AutoCloseable {
 
   public HiveSyncTool(Properties props, Configuration hadoopConf) {
     super(props, hadoopConf);
-    String metastoreUris = props.getProperty(METASTORE_URIS.key());
-    // Give precedence to HiveConf.ConfVars.METASTOREURIS if it is set.
-    // Else if user has provided HiveSyncConfigHolder.METASTORE_URIS, then set that in hadoop conf.
-    if (isNullOrEmpty(hadoopConf.get(HiveConf.ConfVars.METASTOREURIS.varname)) && nonEmpty(metastoreUris)) {
-      LOG.info(String.format("Setting %s = %s", HiveConf.ConfVars.METASTOREURIS.varname, metastoreUris));
-      hadoopConf.set(HiveConf.ConfVars.METASTOREURIS.varname, metastoreUris);
+    String configuredMetastoreUris = props.getProperty(METASTORE_URIS.key());
+    String existingHadoopConfMetastoreUris = hadoopConf.get(HiveConf.ConfVars.METASTOREURIS.varname);
```

Comment:
so `props` has higher precedence than `hadoopConf` when there is a conflict on metastore uris, can we simplify it as

```java
    if (nonEmpty(props.getProperty(METASTORE_URIS.key()))) {
      hadoopConf = new Configuration(originalHadoopConf);
      hadoopConf.set(HiveConf.ConfVars.METASTOREURIS.varname, metastoreUris);
    }
```

Reply:
nope. its nuanced. we don't want to create new instances of hadoop conf (new 
Configuration) unnecessarily. thats the intention behind 3 conditional blocks.

Reply:
instantiating hive sync tool with a new hadoop conf is not a perf/memory sensitive path, isn't it? i think the cost of new Configuration() should be negligible in this scenario so simplifying logic should be preferred over minor optimizations

Reply:
@the-other-tim-brown : Can you comment here based on your exp.

Reply:
The copies of the hadoop configuration can impact memory usage on the driver, especially if you are running a multi table delta streamer with high concurrency. If we think this case is not common, we can go with the simpler approach.

Reply:
so wrote a small snippet to check how much memory it could cost. 

```java
    Configuration originalConf = hadoopConf();
    long freeMem0 = Runtime.getRuntime().freeMemory();
    IntStream.range(0, 1000).forEach(i -> {
      originalConf.set("typical.hadoop.configuration.key" + i, "https://www.example.com:8080/path?query=value#fragment" + i);
    });
    System.out.println("no conf entries: " + originalConf.size());
    List<Configuration> l = new ArrayList<>();
    IntStream.range(0, 100).forEach(i -> {
      l.add(new Configuration(originalConf));
    });
    long freeMem1 = Runtime.getRuntime().freeMemory();
    System.out.println("after copy, used mem (MB): " + (freeMem0 - freeMem1) / (1024.0 * 1024.0));
```

each hadoop conf has 2k+ properties and making 100 copies cost 30mb

```
no conf entries: 2162
after copy, used mem (MB): 29.616012573242188
```

this is even extreme case with this number of confs and metasync tasks. so i don't think memory will be an issue.

Reply:
appreciate it. agree then. we can simplify.